### PR TITLE
feat(rust): Multiprocessing with Python message processors, second attempt

### PR DESCRIPTION
**Pull request by untitaker**
reapply https://github.com/getsentry/snuba/pull/4655, the only notable change here is that with processes=1, there is no process being spawned

### CODE REVIEW

File: snuba/cli/rust_consumer.py [hunk 1]
```diff
@@ -84,12 +84,15 @@
...
 )
 @click.option(
-    "--no-skip-write",
-    "no_skip_write",
-    is_flag=True,
-    help="Writes to ClickHouse.",
+    "--skip-write/--no-skip-write",
+    "skip_write",
+    help="Skip the write to clickhouse",
     default=False,
```
Comment (lynnagara) [inline snuba/cli/rust_consumer.py:90]: Does this need to get flipped too?

Comment (untitaker) [reply snuba/cli/rust_consumer.py:90]: ...good catch

**Review by lynnagara — APPROVED** [review]

File: rust_snuba/src/strategies/python.rs [hunk 2]
```diff
@@ -11,133 +15,178 @@ use crate::types::BytesInsertBatch;
...
-        let py_process_message = Python::with_gil(|py| -> PyResult<Py<PyAny>> {
-            let fun: Py<PyAny> = PyModule::from_code(py, &code, "", "")?
-                .getattr("_wrapped")?
-                .into();
-            Ok(fun)
-        })?;
+        let processing_pool = if processes <= 1 {
+            Some(procspawn::Pool::builder(processes)
+                .env("RUST_SNUBA_PROCESSOR_MODULE", python_module)
+                .env("RUST_SNUBA_PROCESSOR_CLASSNAME", python_class_name)
```
Comment (lynnagara) [inline rust_snuba/src/strategies/python.rs:52]: Can we get rid of these eventually? Would be cool if the python strategy could be agnostic of Snuba

Comment (untitaker) [reply rust_snuba/src/strategies/python.rs:52]: i think if we move this out of rust_snuba there will have to be a rethinking of the entire approach

**Pull request merged**